packages/core/src/trigger_api/__test__/keep_fresh.triggers.test.ts

```diff
@@ -214,4 +214,46 @@ describe('keepFresh, triggers as TriggerProtocol', () => {
     // 1 initial + 1 after enabling
     expect(setupListener).toBeCalledTimes(2);
   });
+
+  test('call teardown after config disabling and call start again after enabling', async () => {
+    const trigger = {
+      setup: createEvent(),
+      teardown: createEvent(),
+      fired: createEvent(),
+    };
+
+    const $enabled = createStore(true);
+
+    const teardownListener = vi.fn();
+    trigger.teardown.watch(teardownListener);
```

Comment:
🤓 can we replace `.watch` with `createWatch` from effector package? I will allow us to write `.concurrent` tests in the future.